src/Nethermind/Nethermind.TxPool/Collections/PersistentBlobTxDistinctSortedPool.cs

```diff
@@ -0,0 +1,119 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Nethermind.Core;
+using Nethermind.Core.Caching;
+using Nethermind.Core.Crypto;
+using Nethermind.Logging;
+
+namespace Nethermind.TxPool.Collections;
+
+public class PersistentBlobTxDistinctSortedPool : BlobTxDistinctSortedPool
+{
+    private readonly ITxStorage _blobTxStorage;
+    private readonly LruCache<ValueKeccak, Transaction> _blobTxCache;
+    private readonly ILogger _logger;
+
+    public PersistentBlobTxDistinctSortedPool(ITxStorage blobTxStorage, ITxPoolConfig txPoolConfig, IComparer<Transaction> comparer, ILogManager logManager)
+        : base(txPoolConfig.PersistentBlobStorageSize, comparer, logManager)
+    {
+        _blobTxStorage = blobTxStorage ?? throw new ArgumentNullException(nameof(blobTxStorage));
+        _blobTxCache = new(txPoolConfig.BlobCacheSize, txPoolConfig.BlobCacheSize, "blob txs cache");
+        _logger = logManager.GetClassLogger();
+
+        RecreateLightTxCollectionAndCache(blobTxStorage);
+    }
+
+    private void RecreateLightTxCollectionAndCache(ITxStorage blobTxStorage)
+    {
```

Comment:
It could slow down our client restart. It would be good to check time of restart and memory usage with full blob db.

Reply:
Loading time on my local machine with full blob db (16k) and avg 2.1 blobs per tx is around 10-11s

Reply:
I would really consider loading this in a separate parallel task. We should have as optimal restarts as possible

Reply:
Refactored to use only column with light blob txs instead of full ones - now recreating light collection takes max 100-150ms so not a problem anymore